types/order.d.ts

```diff
@@ -0,0 +1,71 @@
+export interface Options {
+    readonly fields?: Array<string>,
+    readonly eager?: Boolean,
+    readonly eagerL?: Boolean,
+    readonly map?: Boolean,
+    readonly rules?: Boolean,
+    readonly meta?: Boolean,
+    readonly build?: Boolean,
+    readonly fill?: Boolean,
+    readonly resolveA?: Boolean,
+    readonly skip?: Number,
+    readonly limit?: Number,
+    readonly sort?: Array<Object>,
+    readonly raiseE?: Boolean,
+}
+
+export enum Status {
+    created = "created",
+    waitingPayment = "waiting_payment",
+    paid = "paid",
+    sent = "sent",
+    receivd = "received",
+    returned = "returned",
+    canceled = "canceled",
+    completed = "completed"
+}
+
+export interface Order {
+    readonly id: string;
+    readonly user: string;
+    readonly name: string;
+    readonly shippingAddress: string;
+    readonly billingAddress: string;
+    readonly productQuery: string;
+    readonly amount: number;
+    readonly currency: string;
+    readonly status: Status;
+    readonly created: number;
+    readonly modified: number;
+    readonly meta: Record<string, unknown>;
+}
+
+export interface OrderCreate {
+    readonly user?: string;
+    readonly name?: string;
+    readonly shippingAddress?: string;
+    readonly billingAddress?: string;
+    readonly productQuery?: string;
+    readonly amount?: number;
+    readonly currency?: string;
+    readonly status?: Status;
+}
+
+export interface OrderPatch {
+    readonly user?: string;
+    readonly name?: string;
+    readonly shippingAddress?: string;
+    readonly billingAddress?: string;
+    readonly productQuery?: string;
+    readonly amount?: number;
+    readonly currency?: string;
+    readonly status?: Status;
+}

```

Comment:
Please confirm if the required fields (`?`) are aligned with what there is in the model:
- Order and OrderCreate must have the same optional fields, which are the non-mandatory of the Order model
- OrderPath is all optional

Reply:
My bad, there were some inconsistencies, and the email field was missing. I double-checked, everything's consistent. The id, created, modified, meta, currency and status are optional in OrderCreate because they are assigned a default value if not provided. I made them non-optional in Order because they are always set and always provided.

Reply:
That's the way to go 👍